workspaces/vscode-python/Dockerfile

```diff
@@ -0,0 +1,140 @@
+# Enhanced code-server workspace template for PrairieLearn.
+# As discussed here: https://github.com/PrairieLearn/PrairieLearn/issues/3170
+
+FROM --platform=linux/amd64 codercom/code-server:4.20.1-bookworm
+
+# On PL, we want to standardize on using 1001:1001 for the user. We change the
+# "coder" account's UID and GID here so that fixuid has no work to do later.
+# This speeds up container loading time drastically and avoids timeouts.
+USER root
+RUN export OLD_UID=$(id -u coder) && \
+	export OLD_GID=$(id -g coder) && \
+	export NEW_UID=1001 && \
+	export NEW_GID=1001 && \
+	groupmod -g 1001 coder && \
+	usermod -u 1001 -g 1001 coder && \
+	find /home -user $OLD_UID -execdir chown -h $NEW_UID {} + && \
+	find /home -group $OLD_GID -execdir chgrp -h $NEW_GID {} + && \
+	unset OLD_UID OLD_GID NEW_UID NEW_GID
+	
+# Install Python via miniconda
+USER coder
+RUN curl -sfLO https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh \
+ && /bin/bash Miniconda3-latest-Linux-x86_64.sh -b -p /home/coder/miniconda \
+ && PATH="/home/coder/miniconda/bin:$PATH" \
+ && /home/coder/miniconda/condabin/conda init bash \
+ && conda config --add channels conda-forge \
+ && conda install ipython \
+ && conda install pandas \ 
+ && conda install seaborn \
+ && conda install pillow
```

Comment:
These should be pinned to specific versions and put in a separate `requirements.txt` file so that Dependabot can find them. You should also ensure that you add the resulting `requirements.txt` file to our Dependabot config: https://github.com/PrairieLearn/PrairieLearn/blob/master/.github/dependabot.yml

I'd encourage you to match the dependencies and versions included in `prairielearn/grader-python` and `prairielearn/workspace-jupyterlb-python`:

- https://github.com/PrairieLearn/PrairieLearn/blob/033fb69bf120d5747334ee12116e973de0012873/graders/python/requirements.txt
- https://github.com/PrairieLearn/PrairieLearn/blob/033fb69bf120d5747334ee12116e973de0012873/workspaces/jupyterlab-python/requirements.txt

Reply:
I don't think PL workspaces uses conda right? Maybe I will remove conda here